covariance/covarianceXsec.cpp

```diff
@@ -121,6 +122,19 @@ const std::vector<std::string> covarianceXsec::GetSplineParsNamesFromDetID(const
   return returnVec;
 }
 
+const std::vector<SplineInterpolation> covarianceXsec::GetSplineInterpolationFromDetID(const int DetID) {
+  std::vector<SplineInterpolation> returnVec;
+  for (auto &pair : _fSystToGlobalSystIndexMap[SystType::kSpline]) {
+    auto &SplineIndex = pair.first;
+    auto &SystIndex = pair.second;
+    if ((GetParDetID(SystIndex) & DetID )){
+      returnVec.push_back(SplineParams.at(SplineIndex)._SplineInterpolationType);
+    }
+  }
+
+  return returnVec;
+}
+
```

Comment:
Can you try using AppliesToDetID?
https://github.com/mach3-software/MaCh3/blob/1bf48ff1bcaec1c9c15cb9d30057a9312f9dc3ad/covariance/covarianceXsec.cpp#L223C9-L223C23
We gonna have to change det id treamtent and usng function will make process easier

Reply:
yep I'll give it a try